src/components/lib/cos_defcompinfo_api.c

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2016, Phani Gadepalli and Gabriel Parmer, GWU, gparmer@gwu.edu.
+ *
+ * This uses a two clause BSD License.
+ */
+
+#include <cos_defcompinfo_api.h>
+
+static struct cos_defcompinfo curr_defci;
+
+struct cos_defcompinfo *
+cos_defcompinfo_curr_get(void)
+{ return &curr_defci; }
```

Comment:
damn.  We'd really like the checking so that we can help people use the API correctly.  Perhaps add checks that the `_curr_get` returns an initialized value when it is used within this API?  These checks would likely be added into the child component creation function.  Perhaps elsewhere.